### ISSUE TITLE
Add Avatar for channels

### ISSUE THREAD

**Issue by bazarnew** [RocketChat/Rocket.Chat#3582]
Your Rocket.Chat version: (make sure you are running the latest)
HI
The proposal for channels
Avatar set to channels that have more beauty
Example: See photos
![qqqdd](https://cloud.githubusercontent.com/assets/13213506/16156260/ab8f588c-34c8-11e6-9548-83039bbab1ec.png)

**Comment by Nemra1** [comment]
i love it....i hope to get that soon

**Comment by hniemann** [comment]
+1

**Comment by RemizZ** [comment]
+1

**Comment by Nonymus** [comment]
+1

**Comment by djschilling** [comment]
I also would like to have the opportunity to add avatars/images to channels.
Currently the channels are really collerful and i cannot distinguish good enough the different channels.

Choosing an channel avatar would really improve that.

**Comment by otakuu** [comment]
+1

**Comment by CBrunsch** [comment]
+1

**Comment by phantom603** [comment]
+1

**Comment by c0dzilla** [comment]
Taking this up :smiley:

**Comment by cedrikv** [comment]
+1

**Comment by mwlang** [comment]
+1

**Comment by kvaps** [comment]
Please don't noise by leaving `+1` comments, use github reactions instead :+1:

**Comment by mwlang** [comment]
hmmm...thats a new feature to me.  +1 on an issue subscribes me to the issue and I can begin tracking, perhaps even contributing a PR if I figure out how to do it.  Does the thumbs up feature auto-subcribe one to the updates on the issues?

**Comment by graywolf336** [comment]
@mwlang it doesn't auto subscribe to this issue, however using the reactions plus hitting the subscribe button works just as good and doesn't send notifications to everyone - as we do pay attention to the activity and if someone comments something other than `+1`, it could be useful. :)

**Comment by szab100** [comment]
+1

**Comment by Radeq** [comment]
+1 It would be nice

**Comment by magik092** [comment]
+1

**Comment by Timer91** [comment]
Hey,

Is there any predicted date for this issue ?

Thx

**Comment by Xerionty** [comment]
I'd also love to see avatars on (private) channels.

**Comment by meisterfuu** [comment]
This would be very nice to improve the channel list clarity

**Comment by Agisbert** [comment]
We want this! +1 :)

**Comment by keithhenning** [comment]
+1

**Comment by lindalindap10** [comment]
+1

**Comment by marz71** [comment]
+1

**Comment by fjakop** [comment]
+1

**Comment by juzejunior** [comment]
+1

**Comment by johanneszink** [comment]
+1

**Comment by forresthopkinsa** [comment]
Can someone in charge please delete all these +1 comments?

**Comment by localguru** [comment]
+1 :)

**Comment by SteveUNSW** [comment]
+1 💃

**Comment by appelmoesje** [comment]
Would love this feature!

**Comment by Marcel-123** [comment]
this would be very helpful

**Comment by dsroot** [comment]
We are waiting, but he is not ((((

**Comment by igor-binaryanvil** [comment]
+1

**Comment by CXgamer** [comment]
+1

**Comment by fifou17** [comment]
+1

**Comment by srpatel** [comment]
In case anyone is interested in looking into this, here's what I've found so far:

* Avatars for channels are loaded from a url like: `/avatar/%40channel-name?_dc=0` (`%40` being an encoded `@`)
* The endpoint for getting avatars is at `Rocket.Chat/server/startup/avatar.js`

In `avatar.js`, I believe we'd just need to add an alternate case for the `if (username[0] !== '@') { ... }` condition. I'm not sure what the best way would be to store room "avatars", but if anyone has a good solution I could take a look at implementing this.

**Comment by marz71** [comment]
Dear fellow rocket chatters!
This problem can be solve quite easy. You need some programming skills, but not much.

If using apache add this line to your VirtualHost configuration, like this:

RewriteRule   "^/avatar/@(.*)"  "/avatar.php?p=$1"  [PT]

avatar.php can look like this:


`<?php
  $urlArr = parse_url($_SERVER['REQUEST_URI']);
  $pathArr = explode("/",$urlArr['path']);

  if (isset($pathArr[2])) {
    $roomName = str_replace("%40","",$pathArr[2]);
  }
  $roomNameArr = array(
    'room-a' => 1,
    'room-b' => 2,
    'room-c' => 3
  );
  $id = 0;
  if (isset($roomNameArr[$roomName])) {
    $id = $roomNameArr[$roomName];
  }
 if ($id == 0) {
    $host = $_SERVER['HTTP_HOST'];
    $uri = str_replace('@','',$_SERVER['SCRIPT_URI']);

    $opts = array(
      'http'=>array(
          'method'=>"GET",
      )
    );
    $context = stream_context_create($opts);

    header("Access-Control-Allow-Origin: *");
    header("Content-Type: image/svg+xml");

    echo file_get_contents($uri, false, $context);
    exit;
  }
  $size = 290;
  $file = "gfx/avatars/".$size."x".$size."/avatar".$size."-".$id.".png";

  header("Content-type: image/png");
  header("Accept-Ranges: bytes");
  header('Content-Length: ' . filesize($file));
  header("Last-Modified: Fri, 03 Mar 2004 06:32:31 GMT");
  readfile($file);
?>`


This way you can add custom avatars to your rooms. Hope this can lead you on a right way. You probably need to adjust the .php file to your needs and place it in right place according to your configuration.

**Comment by tmiveda** [comment]
@marz71 
Thanks for the suggestion of this solution.

This may be a solution if the operator of the rocket.chat installation and the administrator of the chat are one and the same person. 
We seperated these responsibilities so when an administrator adds a new channel he has to contact the operator to add a new icon and adjust the script.

For private channels it is even worse since in our organisation anybody can create private channels. 

So yes, this solution may work for some of us but does not reduce the usefulness of this feature request.

**Comment by thawn** [comment]
@marz71 you seem to have dived into this quite a bit and you also seem to have the necessary programming skills. Any chance, you could adapt the rocket.chat source code to implement this feature and create a pull request?

**Comment by tea-mo903** [comment]
Besides the endless +1 comments, I'd also like to see that feature. That would make our endlessy growing amount of RC channels more easy to distinguish between

![rc_channels](https://user-images.githubusercontent.com/11437666/56656341-8e980b00-6695-11e9-8da9-9005d3f1b104.png)

**Comment by Agisbert** [comment]
### +1
Having multiple channels is definitely a reason

**Comment by Xarkam** [comment]
Hello, this is a css workaround based on aria-label. If you change the channel name, you need to change the value of aria-label in your custom css.

The result:
![image](https://user-images.githubusercontent.com/331816/57154200-96178c80-6dd8-11e9-86f5-6da27779568a.png)

How to:
in rocket.chat admin, goto appearance -> custom css
for each channel add
```css
a[aria-label="general"] img {
  border: 3px orange solid;
}
```
Simply change `"general"` between double quote marks by the name of the channel.

And save.

**Note**: It will not work with a browser too old

**Comment by joshmartintv** [comment]
@Xarkam This is a nice workaround, but would not allow the proper separation of Administrator and User roles.  Your solution would require the user to contact the admin to update css for each new channel which introduces more work for the administrator.  The end user should be able to set an avatar after they create their channel without the need to contact the administrator.

**Comment by qiangzhao** [comment]
+1 👍

**Comment by tmiveda** [comment]
Ich kehre zurück am 21.05.2019.

Bitte wenden Sie sich in dringenden Fällen an den Support:
http://www.veda.net/customer-service
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
17.05.2019 8:20:08.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by qiangzhao** [comment]
> Ich kehre zurück am 21.05.2019. Bitte wenden Sie sich in dringenden Fällen an den Support: http://www.veda.net/customer-service oder zentral an die Nummer 02404/5507 0. Viele Grüße, Torsten Mingers Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht "Re: [RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am 17.05.2019 8:20:08. Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während diese Person abwesend ist.

When I trying to translate in Google, I nearly can't catch anything your meanings. So I think maybe comment in English will help foreigners easier to understand. :-)

**Comment by Maqsyo** [comment]
> When I trying to translate in Google, I nearly can't catch anything your meanings. So I think maybe comment in English will help foreigners easier to understand. :-)

This is just an autoresponder. So it hasn't any meaning for this issue 😁 
It means, that this person isn't reachable at the moment and you could call the office-number if it is really important 😂

**Comment by qiangzhao** [comment]
> > When I trying to translate in Google, I nearly can't catch anything your meanings. So I think maybe comment in English will help foreigners easier to understand. :-)
> 
> This is just an autoresponder. So it hasn't any meaning for this issue grin
> It means, that this person isn't reachable at the moment and you could call the office-number if it is really important joy

Okay, then, Thanks for the explain, so appreciate!

**Comment by tmiveda** [comment]
Ich kehre zurück am 21.05.2019.

Bitte wenden Sie sich in dringenden Fällen an den Support:
http://www.veda.net/customer-service
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
20.05.2019 9:43:35.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by tmiveda** [comment]
Ich kehre zurück am 21.05.2019.

Bitte wenden Sie sich in dringenden Fällen an den Support:
http://www.veda.net/customer-service
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
20.05.2019 9:31:02.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by tmiveda** [comment]
Ich kehre zurück am 21.05.2019.

Bitte wenden Sie sich in dringenden Fällen an den Support:
http://www.veda.net/customer-service
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
20.05.2019 9:18:05.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by merskip** [comment]
So on every comment on this thread, we'll get a new free comment form him :D

**Comment by Agisbert** [comment]
@tmiveda, thanks for keeping this feature alive! We really needed a strong supported like you.

This feature will help us both organize our ginormous channels as well as boost our developers mood.

Bring this soon!

**Comment by Maqsyo** [comment]
So Rocket.Chat only has time until the 21.05.2019 to develop this feature, because @tmiveda will be back then. Right?

**Comment by merskip** [comment]
@mass10 Yes! We don't want for @tmiveda will back to the office and see the feature isn't done.

**Comment by tmiveda** [comment]
Ich kehre zurück am 21.05.2019.

Bitte wenden Sie sich in dringenden Fällen an den Support:
http://www.veda.net/customer-service
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
20.05.2019 10:53:57.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by tmiveda** [comment]
Ich kehre zurück am 21.05.2019.

Bitte wenden Sie sich in dringenden Fällen an den Support:
http://www.veda.net/customer-service
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
20.05.2019 11:43:26.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by tmiveda** [comment]
Ich kehre zurück am 21.05.2019.

Bitte wenden Sie sich in dringenden Fällen an den Support:
http://www.veda.net/customer-service
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
20.05.2019 10:43:41.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by tmiveda** [comment]
Ich kehre zurück am 21.05.2019.

Bitte wenden Sie sich in dringenden Fällen an den Support:
http://www.veda.net/customer-service
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
20.05.2019 11:38:59.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by forresthopkinsa** [comment]
@tmiveda please clean up your comments.

**Comment by Agisbert** [comment]
@forresthopkinsa I strongly disagree. No deletion! 

PLUS! @tmiveda is one of the biggest supporters of this feature.

**Comment by tmiveda** [comment]
> @tmiveda please clean up your comments.

Yep. I deleted all "autoreply" messages.  Sorry for the spam!

**Comment by ramrami** [comment]
Here we go #14618 a WIP

<img width="1435" alt="Capture d’écran 2019-05-23 à 19 16 04" src="https://user-images.githubusercontent.com/9569590/58319088-b13e5080-7e08-11e9-863e-db876ae7fac2.png">

**Comment by engelgabriel** [comment]
https://github.com/RocketChat/Rocket.Chat/issues/14621

**Comment by gcommit** [comment]
When will this be published? Waiting for more than one year O.o

**Comment by agisbert** [comment]
I cannot believe that **Microsoft teams** has this feature and the RocketChat Team hasn't take this seriously for more than 3 years. 

This is really disappointing.

**Comment by ankar84** [comment]
When?

**Comment by kromonos** [comment]
> When?

It's done, when it's done.

**Comment by ankar84** [comment]
> It's done, when it's done.

Answer is true, but the meaning in it is zero

**Comment by thawn** [comment]
@ankar84 since this feature appears to be important to you: this project is open source. You can start implementing it right away or hire someone to do it for you. Then submit a pull request to get your implementation into the mainline source.

**Comment by CoMPaTech** [comment]
@thawn 
>  Then submit a pull request to get your implementation into the mainline source.

Just note that there was a PR but the author deleted his branch last week: https://github.com/RocketChat/Rocket.Chat/pull/14618

**Comment by ankar84** [comment]
> @ankar84 @gcommit @agisbert @kromonos since this feature appears to be important to you: this project is open source. You can start implementing it right away or hire someone to do it for you. Then submit a pull request to get your implementation into the mainline source.

Yes, @CoMPaTech right. There **was** PR for that.

**Comment by thawn** [comment]
Apparently noone found the time to do a code review. That is indeed a pity. Although I don't quite understand why the author deleted his PR.

**Comment by knarrff** [comment]
> Although I don't quite understand why the author deleted his PR.

I guess also the author of a PR might get tired of waiting for something else to happen to a PR than adapting milestones from time to time. After all, it is the PR author that has to keep the PR in-line with upstream during all that time. Once the PR isn't important enough to the author anymore, and it is apparently not important to the dev-team as well, the author might as well abandon it and not keep it in their active PR list.

It is not that the code got deleted. It's all still there for anyone to pick up. It's just that the original author apparently didn't want to keep hanging in there for so long.

And yes, this is a pity.

**Comment by reetp** [comment]
An ongoing management & communication problem that has existed for a long time, and lost Rocket a lot of code, developers, and good will.

You'd think they would have got on top of this by now...... clearly not.

**Comment by forresthopkinsa** [comment]
> > @ankar84 @gcommit @agisbert @kromonos since this feature appears to be important to you: this project is open source. You can start implementing it right away or hire someone to do it for you. Then submit a pull request to get your implementation into the mainline source.
> 
> Yes, @CoMPaTech right. There **was** PR for that.

If you're so impatient then just build that PR and use that

**Comment by dimm0** [comment]
It's not even a year since the PR was submitted... So impatient..

**Comment by agisbert** [comment]
> > > @ankar84 @gcommit @agisbert @kromonos since this feature appears to be important to you: this project is open source. You can start implementing it right away or hire someone to do it for you. Then submit a pull request to get your implementation into the mainline source.
> > 
> > 
> > Yes, @CoMPaTech right. There **was** PR for that.
> 
> If you're so impatient then just build that PR and use that

So much hate and anger! Easy going. 

Not that a PR (that already existed) will solve this issue.

**Comment by gcommit** [comment]
Wow that escalated quickly. There was a PR for this FOR A WHOLE YEAR and nothing happend. But since the community is asking for it, you guys start blaming us. Good job 👍

**Comment by thawn** [comment]
It seems that I have contributed to escalating this thread. At the time of writing my comment above I was unaware of the existing pull request. I have edited the comment accordingly and apologize to anyone who felt offended.

**Comment by forresthopkinsa** [comment]
I'm also not a member of the RC team, I'm just annoyed by weekly notifications coming from this thread with obnoxious comments like "+1" and "when?" over and over again.

Should they have reviewed that PR a year ago? Totally. But seeing as they didn't, stop whining about it and appreciate that there *is* a PR, and just go use that code branch. There, you have your channel avatars.

Like thawn said -- it's open-source. This is how open-source works. Sometimes PRs are neglected and go stale. Making a fuss about it might help in some circumstances, but it consistently pisses people off.

I guarantee you that if someone had put some [money](https://www.bountysource.com/teams/rocketchat) behind that PR, it wouldn't have gone stale. Seeing as they didn't, you just have to deal with the time of volunteers.

tl;dr chill out

**Comment by agisbert** [comment]
> I'm also not a member of the RC team, I'm just annoyed by weekly notifications coming from this thread with obnoxious comments like "+1" and "when?" over and over again.
> 
> Should they have reviewed that PR a year ago? Totally. But seeing as they didn't, stop whining about it and appreciate that there _is_ a PR, and just go use that code branch. There, you have your channel avatars.
> 
> Like thawn said -- it's open-source. This is how open-source works. Sometimes PRs are neglected and go stale. Making a fuss about it might help in some circumstances, but it consistently pisses people off.
> 
> I guarantee you that if someone had put some [money](https://www.bountysource.com/teams/rocketchat) behind that PR, it wouldn't have gone stale. Seeing as they didn't, you just have to deal with the time of volunteers.
> 
> tl;dr chill out

Dude, really? Notifications annoy you? Here is a small tip -> 

<img width="580" alt="image" src="https://user-images.githubusercontent.com/5805538/73365555-be231600-42ac-11ea-9313-4ec953e9ef67.png">

+1 is a valid way of showing interest. I don't know why we need multi paragraph comments only.

I remember the auto reply messages some months ago. That made my week! I really have fun with this thread, plus, it exposes the slowness of RC changes.

**Comment by forresthopkinsa** [comment]
There have been valuable contributions in this thread, but you wouldn't know it at a glance, because they're lost in a sea of useless comments.

This discussion right now is also irrelevant, and is sending the thread even further off-topic. I'm glad you're having fun, but diluting the thread with all this garbage only makes it *harder* to track the issue you're so passionate about.

**Comment by agisbert** [comment]
> There have been valuable contributions in this thread, but you wouldn't know it at a glance, because they're lost in a sea of useless comments.
> 
> This discussion right now is also irrelevant, and is sending the thread even further off-topic. I'm glad you're having fun, but diluting the thread with all this garbage only makes it _harder_ to track the issue you're so passionate about.

Yet you keep on answering. I think you love the discussion too 😉 😘 ...

**Comment by agentlibre** [comment]
In the meantime, is there any option to display no avatar for the channels? Otherwise, this could be another option to add.

**Comment by TDzyne** [comment]
https://forums.rocket.chat/t/channel-avatar-generator/5882

**Comment by Xerionty** [comment]
> https://forums.rocket.chat/t/channel-avatar-generator/5882

Okay and how do we use this?

**Comment by johnnydement** [comment]
Also notice these generator icons don't work in phone.

**Comment by evilrussian** [comment]
any updates?

**Comment by tmiveda** [comment]
Ich kehre zurück am 15.07.2020.

Bitte wenden Sie sich in dringenden Fällen an den Support:
https://www.veda.net/support-center/
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
08.07.2020 20:27:56.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by agisbert** [comment]
@tmiveda, thanks for you support.

**Comment by tmiveda** [comment]
Ich kehre zurück am 15.07.2020.

Bitte wenden Sie sich in dringenden Fällen an den Support:
https://www.veda.net/support-center/
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
09.07.2020 7:57:53.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by Maqsyo** [comment]
Oh! He's on vacation again!
Thanks @tmiveda again for Spur Support!

**Comment by tmiveda** [comment]
Ich kehre zurück am 15.07.2020.

Bitte wenden Sie sich in dringenden Fällen an den Support:
https://www.veda.net/support-center/
oder zentral an die Nummer 02404/5507 0.

Viele Grüße,
Torsten Mingers


Hinweis: Dies ist eine automatische Antwort auf Ihre Nachricht  "Re:
[RocketChat/Rocket.Chat] Add Avatar for channels (#3582)" gesendet am
11.07.2020 9:33:58.

Diese ist die einzige Benachrichtigung, die Sie empfangen werden, während
diese Person abwesend ist.

**Comment by lelvisl** [comment]
3.5.0 already released. may be add this feature in 3.6.0?

**Comment by frdmn** [comment]
@lelvisl You are right - it'll drop with 3.6.0:  https://github.com/RocketChat/Rocket.Chat/pull/18443

**Comment by ggazzo** [comment]
closed by https://github.com/RocketChat/Rocket.Chat/pull/18443
